libminifi/src/utils/StringUtils.cpp

```diff
@@ -538,6 +538,56 @@ nonstd::expected<std::optional<char>, ParseError> parseCharacter(std::string_vie
   return nonstd::make_unexpected(ParseError{});
 }
 
+std::string replaceEscapedCharacters(std::string_view input) {
+  std::stringstream result;
+  for (size_t i = 0; i < input.size(); ++i) {
+    char input_char = input[i];
+    if (input_char != '\\' || i == input.size() - 1) {
+      result << input_char;
+      continue;
+    }
+    char next_char = input[i+1];
+    switch (next_char) {
+      case '0':
+        result << '\0';  // Null
+        ++i;
+        break;
+      case 'a':
+        result << '\a';  // Bell
+        ++i;
+        break;
+      case 'b':
+        result << '\b';  // Backspace
+        ++i;
+        break;
+      case 't':
+        result << '\t';  // Horizontal Tab
+        ++i;
+        break;
+      case 'n':
+        result << '\n';  // Line Feed
+        ++i;
+        break;
+      case 'v':
+        result << '\v';  // Vertical Tab
+        ++i;
+        break;
+      case 'f':
+        result << '\f';  // Form Feed
+        ++i;
+        break;
+      case 'r':
+        result << '\r';  // Carriage Return
+        ++i;
+        break;
+      default:
+        result << '\\';
```

Comment:
This is an interesting question: should `"\s"` expand to `"s"` or `"\s"`?  The old `parseCharacter` function does the first, here you do the second. I think it's a bit confusing that they are different, but I can be persuaded either way.